src/AlgebraicGeometry/Surfaces/K3Auto.jl

```diff
@@ -2040,3 +2044,66 @@ function has_zero_entropy(S::ZZLat; rank_unimod=26)
 
   return maximum(push!([sign(i) for i in d],-1)), data, K3Autgrp, chambers, rational_curves
 end
+
+  
+function _get_pos_entropy_element(gensAutL)
+  G = [first(gensAutL)^0]
+  f = first(gensAutL)^0
+  n = nrows(f)
+  n<= 4 || error("not implemented")
+  while true
+    Gnew = []
+    for g in G
+      for h in gensAutL
+        f = g*h
+        push!(Gnew, f)
+        if rank(f^12-1)==2   # positive entropy
+          return f
+        end
+      end
+    end
+    G = Gnew
+  end
+end
+
+function is_virtually_abelian(L::ZZLat, gensAutL)
+  length(gensAutL)==0 && return true
+  rank(L)>= 5 && return has_zero_entropy(L, gensAutL)
+  if has_zero_entropy(L, gensAutL)
+    return true
+  end
+  # search an automorphism of positive entropy.
+  f = _get_pos_entropy_element(gensAutL)
+  K = kernel(f^12-1)
+  k = nrows(K)
+  if k == rank(L)-2
+    for g in gensAutL
+      r = rank(vcat(K,K*g))
+      if r > k
+        return false
+      end
+    end
+    return true
+  end 
+  # now k = 0 
+  @assert k == 0
+  return all(g*f == f*g for g in gensAutL)
+end
+
+function has_zero_entropy(L::ZZLat, gensAutL)
+  length(gensAutL)==0 && return true
+  C = lattice(rational_span(L), _common_invariant(gensAutL)[2])
+  d = diagonal(rational_span(C))
+  return 0<= maximum(push!([sign(i) for i in d],-1))
```

Comment:
Isn't this the same as
```suggestion
  return !all(i -> sign(i) < 0, d)
```
or
```suggestion
  return !all(sign(i) < 0 for i in d)
```
or
```suggestion
  return any(i -> sign(i) >= 0, d)
```

Reply:
what if d is empty?

Reply:
should be o.k. thx.